packages/hw-app-btc/src/Btc.js

```diff
@@ -493,55 +544,82 @@ btc.createPaymentTransactionNew(
               );
             }
           })
-        );
-      })
+        )
+      )
+      .then(() =>
+        doIf(segwit, () =>
+          // Do the first run with all inputs
+          this.startUntrustedHashTransactionInput(
+            true,
+            targetTransaction,
+            trustedInputs,
+            segwit
+          ).then(() =>
+            doIf(!resuming && typeof changePath != "undefined", function() {
+              return this.provideOutputFullChangePath(changePath);
+            }).then(() => this.hashOutputFull(outputScript))
+          )
+        )
+      )
       .then(() =>
+        // Do the second run with the individual transaction
         foreach(inputs, (input, i) => {
           targetTransaction.inputs[i].script =
             inputs[i].length >= 3 && typeof inputs[i][2] === "string"
               ? Buffer.from(inputs[i][2], "hex")
-              : regularOutputs[i].script;
+              : !segwit
+                ? regularOutputs[i].script
+                : Buffer.concat([
+                    Buffer.from("76a914", "hex"),
+                    this.hashPublicKey(publicKeys[i]),
+                    Buffer.from("88ac", "hex")
```

Comment:
what are these constants? we could put them statically just so we actually have them named?

Reply:
These are operators for the spending script. I' ll have them named